sphinx/builders/html/__init__.py

```diff
@@ -240,6 +242,32 @@ def init(self) -> None:
 
         self.use_index = self.get_builder_config('use_index', 'html')
 
+    def merge_builder_post_transform(self, new_attrs: dict[str, Any]) -> None:
+        """Merge images and search indexer back to the main builder after parallel
+        post-transformation.
+
+        :param new_attrs: the attributes from the parallel subprocess to be
+                          udpated in the main builder (self)
+        """
+        # handle indexer
+        if self.indexer is None:
+            lang = self.config.html_search_language or self.config.language
+            self.indexer = IndexBuilder(self.env, lang,
+                                        self.config.html_search_options,
+                                        self.config.html_search_scorer)
+        indexer_data = new_attrs['indexer']
```

Comment:
With this, I highly think we should use "context" instead of "new_attrs". Because it's more like an "update" rather than replacing one attribute with another.